src/experimental/tools/mediaCapabilitiesProber/api.ts

```diff
@@ -0,0 +1,191 @@
+  IMediaKeySystemConfiguration,
+} from "./types";
+
+/**
+ * A set of API to probe media capabilites.
+ * Each API allow to evalute a specific feature (HDCP support, decoding infos, etc)
+ * and relies on different browser API to probe capabilites.
+ */
+const mediaCapabilitiesProber = {
+  /**
+   * Set logger level
+   * @param {string} level
+   */
+  set LogLevel(level: string) {
+    log.setLevel(level, log.getFormat());
+  },
+
+  /**
+   * Get logger level
+   * @returns {string}
+   */
+  get LogLevel(): string {
+    return log.getLevel();
+  },
+
+  /**
+   * @param {string} keySystemType - Reverse domain name of the wanted key
+   * system.
+   * @param {Array.<MediaKeySystemConfiguration>} keySystemConfiguration - DRM
+   * configuration wanted.
+   * @returns {Promise.<MediaKeySystemConfiguration>} - Resolved the
+   * MediaKeySystemConfiguration actually obtained if the given configuration is
+   * compatible or a rejected Promise if not.
+   */
+  async checkDrmConfiguration(
+    keySystemType: string,
+    keySystemConfiguration: IMediaKeySystemConfiguration[],
+  ): Promise<MediaKeySystemConfiguration> {
+    if (isNullOrUndefined(eme.requestMediaKeySystemAccess)) {
+      const error = new Error("EME not supported in current environment");
+      throw error;
+    }
+
+    const mksa = await eme.requestMediaKeySystemAccess(
+      keySystemType,
+      keySystemConfiguration,
+    );
+    return mksa.getConfiguration();
+  },
+
+  /**
+   * Get HDCP status. Evaluates if current equipement support given
+   * HDCP revision.
+   * @param {string} hdcpVersion - The wanted HDCP version to test (e.g.
+   * `"1.1"`).
+   * @returns {Promise.<string>} - Returns a Promise which rejects if the
+   * browser API we need to rely on unexpectedly fail.
+   *
+   * If they resolve, resolve with one of the following string:
+   *
+   *   - `"Supported"`: The HDCP version is probably supported
+   *
+   *   - `"NotSupported"`: The HDCP version is probably not supported
+   *
+   *   - `"Unknown"`: It is unknown if the HDCP version may be supported or not.
+   */
+  async getStatusForHDCP(hdcpVersion: string): Promise<string> {
+    if (hdcpVersion === undefined || hdcpVersion.length === 0) {
+      throw new Error("Bad Arguments: " + "No HDCP Policy specified.");
+    }
+    try {
+      const res = await probeTypeWithFeatures({ hdcp: hdcpVersion });
+      switch (res) {
+        case "NotSupported":
+        case "Supported":
+          return res;
+        case "Unknown":
+          // continue
+          break;
+      }
+    } catch (err) {
+      // We do not care about this call erroring
+    }
+    return probeHDCPPolicy(hdcpVersion);
+  },
+
+  /**
+   * Get decoding capabilities from a given video and/or audio
+   * configuration.
+   * TODO check alongside key system.
+   * @param {Object} mediaConfig
+   * @returns {Promise.<string>}
+   */
+  async getDecodingCapabilities(mediaConfig: IMediaConfiguration): Promise<string> {
```

Comment:
don't the type should be narrowed to "Supported" / "Unknown" / "NotSupported" instead of being a string ?